src/shelter/use-shelter.hook.ts

```diff
@@ -1,39 +1,54 @@
 import { useEffect, useMemo } from 'react';
 import { Alert } from 'react-native';
 import { useDispatch } from 'react-redux';
-import { Subject } from 'rxjs';
+import { merge, Subject } from 'rxjs';
 import { switchMap } from 'rxjs/operators';
 
-import { ImportAccountPayload, importWalletActions } from '../store/wallet/wallet-actions';
+import { ScreensEnum } from '../navigator/screens.enum';
+import { useNavigation } from '../navigator/use-navigation.hook';
+import { addHdAccount } from '../store/wallet/wallet-actions';
+import { useWalletSelector } from '../store/wallet/wallet-selectors';
 import { generateSeed } from '../utils/keys.util';
 import { Shelter } from './shelter';
-import { importWalletOperator$ } from './wallet.util';
 
 export const useShelter = () => {
   const dispatch = useDispatch();
+  const wallet = useWalletSelector();
+  const { navigate } = useNavigation();
 
-  const importWallet$ = useMemo(() => new Subject<ImportAccountPayload>(), []);
+  const importWallet$ = useMemo(() => new Subject<{ seedPhrase: string; password: string }>(), []);
   const createWallet$ = useMemo(() => new Subject<string>(), []);
-  const revealValue$ = useMemo(() => new Subject<string>(), []);
+  const createHdAccount$ = useMemo(() => new Subject<string>(), []);
+  const revealSecretKey$ = useMemo(() => new Subject<string>(), []);
+  const revealSeedPhrase$ = useMemo(() => new Subject(), []);
 
   useEffect(() => {
     const subscriptions = [
       importWallet$
-        .pipe(switchMap(({ seedPhrase, password }) => importWalletOperator$(seedPhrase, password)))
-        .subscribe(publicData => dispatch(importWalletActions.success(publicData))),
-
+        .pipe(switchMap(({ seedPhrase, password }) => Shelter.importHdAccount$(seedPhrase, password)))
+        .subscribe(publicData => dispatch(addHdAccount(publicData))),
       createWallet$.subscribe(password => importWallet$.next({ seedPhrase: generateSeed(), password })),
-
-      revealValue$.pipe(switchMap(key => Shelter.revealValue$(key))).subscribe(value => {
-        value !== undefined && Alert.alert(value, '', [{ text: 'OK' }]);
-      })
+      createHdAccount$
```

Comment:
Investigate errors occurrence and add error handlers

Reply:
Added basic error handling (`undefined` returned in case of error).  
In separate task will add toasts with error